app/feature/feature-insurances/src/test/kotlin/com/hedvig/android/feature/insurances/insurancedetail/ContractDetailPresenterTest.kt

```diff
@@ -0,0 +1,222 @@
+package com.hedvig.android.feature.insurances.insurancedetail
+
+import app.cash.turbine.Turbine
+import arrow.core.Either
+import arrow.core.left
+import arrow.core.right
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isInstanceOf
+import com.hedvig.android.core.common.ErrorMessage
+import com.hedvig.android.data.contract.ContractGroup
+import com.hedvig.android.data.contract.ContractType
+import com.hedvig.android.data.productvariant.ProductVariant
+import com.hedvig.android.feature.insurances.data.InsuranceAgreement
+import com.hedvig.android.feature.insurances.data.InsuranceContract
+import com.hedvig.android.featureflags.flags.Feature
+import com.hedvig.android.featureflags.test.FakeFeatureManager
+import com.hedvig.android.logger.TestLogcatLoggingRule
+import com.hedvig.android.molecule.test.test
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.test.runTest
+import kotlinx.datetime.LocalDate
+import org.junit.Rule
+import org.junit.Test
+
+class ContractDetailPresenterTest {
+  @get:Rule
+  val testLogcatLogger = TestLogcatLoggingRule()
+
+  @Test
+  fun `if termination flow enabled and no termination date show cancel insurance button`() = runTest {
+    val getContractForContractIdUseCase = FakeGetContractForContractIdUseCase()
+    val featureManager = FakeFeatureManager(featureMap = { mapOf(Feature.TERMINATION_FLOW to true) })
```

Comment:
That's totally my bad, but can you see if you can use FakeFeatureManager2 here instead?
The first iteration did not allow for use to do stuff like wait and assert states before the feature manager has managed to return something yet and so on.
I just have never taken the time to remove the FakeFeatureManager from the tests that use it and just use "2" everywhere. And obviously remove the "2" part from them 😂